Documentation/design-docs/first-class-structs.md

```diff
@@ -38,12 +38,14 @@ In order to treat fully-enregisterable struct types as "first class" types in Ry
 * `TYP_SIMD8`, `TYP_SIMD12`, `TYP_SIMD16` and (where supported by the target) `TYP_SIMD32`.
  - These types already exist, and represent some already-completed steps toward First Class Structs.
 
- We propose to create the following additional types to be used where struct types of the given size are
+ We had previously proposed to create additional types to be used where struct types of the given size are
  passed and/or returned in registers:
  * `TYP_STRUCT1`, `TYP_STRUCT2`, `TYP_STRUCT4`, `TYP_STRUCT8` (on 64-bit systems)
 
-For other struct types, for which the JIT has no type-specific optimizations, the following
-transformations need to be supported effectively:
+However, further discussions have suggested that this may not be necessary. Rather, storage decisions
+should largely be deferred to the backend (`Lowering` and register allocation).
```

Comment:
@CarolEidt BTW, once I get in the block op code gen PR I intend to try something simple like - if the block has suitable register size then don't unroll but instead build LSRA def/use ref positions as if the load/store nodes are normal indirs. If that works we can look at extending this to call args.